### PR TITLE
feat: add set_union_with_tombstones

### PR DESCRIPTION
**Pull request by zzlk**

### CODE REVIEW

File: lattices/src/collections.rs
Comment (MingweiSamuel) [inline lattices/src/collections.rs:1]: Nice

File: lattices/src/set_union_with_tombstones.rs [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+use crate::collections::{ArraySet, EmptySet, OptionSet, SingletonSet};
+use crate::{IsBot, IsTop, LatticeFrom, LatticeOrd, Merge};
+
+/// Set-union lattice with tombstones.
+///
+/// When an item is deleted from the SetUnionWithTombstones, it is removed from `set` and added to `tombstones`.
+/// This also is an invariant, if an item appears in `tombstones` it must not also be in `set`.
+///
+/// Merging set-union lattices is done by unioning the keys of both the (set and tombstone) sets,
+/// and then removing any items in `set` that now exist in `tombstones`
```
Comment (MingweiSamuel) [inline lattices/src/set_union_with_tombstones.rs:18]: This sounds like set XOR when it should be set difference?

**Review by MingweiSamuel — APPROVED** [review]
Looks good just confused by the description

File: lattices/src/set_union_with_tombstones.rs (hunk 1 above)
Comment (zzlk) [reply lattices/src/set_union_with_tombstones.rs:18]: hmm, that is not what I'm trying to convey. it should be set subtraction, set = set - tombstones, that is the invariant and operation that is performed.

**Pull request merged**